makeflow/src/makeflow.c

```diff
@@ -1424,6 +1517,15 @@ int main(int argc, char *argv[])
 			remote_jobs_max = MIN(local_jobs_max, n);
 		}
 	}
+	
+	struct dag_node *n;
+	// check to ensure that all jobs can be ran with resources given
+	for(n = d->nodes; n; n = n->next) {
+		if(makeflow_is_local_node(n) && !makeflow_can_alloc_local(n)) {	
+			fprintf(stderr, "Critical error, not enough resources to run job");
```

Comment:
Use the node number in the print statement so that you can print out every node that can not run under the resource limit. Also use a flag to indicate an error occurred and find every instance where a node won't fit and then exit.